test/test_infra.py

```diff
@@ -96,3 +98,32 @@ def test_ensure_copyright(self):
                     issues.append(filepath)
         if issues:
             self.fail("Please add copyright headers to the following files:\n" + "\n".join(issues))
+
+
+class ImportersTestCase(unittest.TestCase):
+
+    template = "from ops import {module_name}"
+
+    def test_imports(self):
+        mod_names = [
+            'charm',
+            'framework',
+            'main',
+            'model',
+            'testing',
+        ]
+
+        for name in mod_names:
+            with self.subTest(name=name):
+                self.check(name)
+
+    def check(self, name):
+        """Helper function to run the test."""
+        _, testfile = tempfile.mkstemp()
+        self.addCleanup(os.unlink, testfile)
+
+        with open(testfile, 'wt', encoding='utf8') as fh:
+            fh.write(self.template.format(module_name=name))
+
+        proc = subprocess.run(['python3', testfile])
```

Comment:
I think you want this to be
subprocess.run([sys.executable,...])

Reply:
Good idea, thanks!